### DISCUSSION TITLE
How to Change HTML Markup Stored in Database

### DISCUSSION THREAD

**Discussion by ruslansteiger** [awcodes/filament-tiptap-editor#416] [Q&A]
Hey everyone 👋

I have a specific requirement for the HTML markup that is stored in the database.

By default, the editor saves buttons in the following format:

```html
<a href="https://www.google.com/" data-as-button="true">Never gonna give you up.</a>
```

However, I need to store the markup in the following format instead:

```html
<div class="button">
  <a href="https://www.google.com/">Never gonna give you up.</a>
</div>
```

Is there a way to customize the Editor so that it saves the button links wrapped in a div with a class of button as shown above? If so, could someone guide me on how to achieve this?

Thank you in advance for your help! 😊

Best regards,
Ruslan

**Comment by thewhaleiam** [comment]
You can leverage handleRecordUpdate on the resource record, something like this

```
protected function handleRecordUpdate(Model $record, array $data): Model
    {
        // generate the correct HTML for storage, preg_replace
        $data['body'] = $this->wrapButtonTagsInBody($data['body']);
        $record->update($data);
        return $record;
    }
```
you will also need to hydrate the body so that returns it to it's original format when loading the body, so that you don't have any undesirably formatting issues.
